src/hotspot/share/gc/shenandoah/shenandoahTaskqueue.hpp

```diff
@@ -169,8 +175,16 @@ class ObjArrayChunkedTask
     return (int) ((val >> pow_shift) & right_n_bits(pow_bits));
   }
 
-  inline uintptr_t encode_oop(oop obj) const {
-    return ((uintptr_t)(void*) obj) << oop_shift;
+  inline uintptr_t encode_oop(oop obj, bool count_liveness, bool strong) const {
+    uintptr_t encoded_oop = ((uintptr_t)(void*) obj) << oop_shift;
+    assert((encoded_oop & (count_liveness_decode_mask | strong_decode_mask)) == 0, "need bit for encoding count-liveness and strong bits");
```

Comment:
Ah! This assert can be sacrificed for the additional asserts in constructors, right?